gitpod-io/gitpod
# [supervisor] terminal API enhancements

#### What it does

This PR completes the supervisor terminal API to implement the remote terminal channel by the Code Server.

Diff in Code: https://github.com/gitpod-io/vscode/compare/001c337aa00d9eff6faa2c07255451d1c33602e0...gp-code

fix #2855, fix #2793, fix #2394

#### How to test

- Start Theia and Code check that regular and task terminals are properly functional, i.e. try to start and shutdown terminals, enter some input, resize them and so on.
- Now in Code: regular terminals should be preserved between page reloads and server restarts.
